tests/test.bats

```diff
@@ -62,107 +68,203 @@ teardown() {
+}
 
-  # Test stat command (should not fail even without auth)
-  run ddev tailscale stat
-  # Command should execute (may show not logged in, but shouldn't crash)
+@test "tailscale url and launch commands exist" {
+  set -eu -o pipefail
+  
+  run ddev add-on get "${DIR}"
+  assert_success
+  
+  run ddev restart -y
+  assert_success
 
-  # Test proxy command
-  run ddev tailscale proxy
-  # Command should execute
+  # Test url command (will fail without auth but should exist)
+  run ddev tailscale url
+  # Don't assert success due to auth requirements
+  
+  # Test launch command structure
+  run timeout 5s ddev tailscale launch --dry-run 2>/dev/null || true
+  # Check command exists, don't worry about success without auth
 }
 
-@test "docker-compose file has required services and volumes" {
+@test "no conflicting processes after restart" {
   set -eu -o pipefail
+  
   run ddev add-on get "${DIR}"
   assert_success
-
-  # Check docker-compose file contains required elements
-  run grep -q "tailscale-router:" ".ddev/docker-compose.tailscale-router.yaml"
+  
+  # First restart
+  run ddev restart -y
   assert_success
-
-  run grep -q "tailscale-router-state:" ".ddev/docker-compose.tailscale-router.yaml"
+  refute_output --partial "foreground already exists"
+  
+  # Second restart (should not have port conflicts)
+  run ddev restart -y  
   assert_success
+  refute_output --partial "foreground already exists"
+  refute_output --partial "ERROR (spawn error)"
+}
 
-  run grep -q "image: \${TS_DOCKER_IMAGE:-tailscale/tailscale:latest}-\${DDEV_SITENAME}-built" ".ddev/docker-compose.tailscale-router.yaml"
+@test "tailscale share command structure" {
+  set -eu -o pipefail
+  
+  run ddev add-on get "${DIR}"
+  assert_success
+  
+  run ddev restart -y
   assert_success
+
+  # Test share command with --help flag
+  run ddev tailscale share --help 2>/dev/null || run ddev tailscale serve --help
+  # One of these should work to show the command is properly structured
 }
 
-@test "configuration supports both private and public modes" {
+@test "configuration files are properly installed" {
   set -eu -o pipefail
+  
   run ddev add-on get "${DIR}"
   assert_success
-
-  # Check private config
-  run grep -q '"AllowFunnel"' ".ddev/tailscale-router/config/tailscale-private.json"
+  
+  # Check all required files are installed
+  run test -f .ddev/commands/host/tailscale
   assert_success
-  run grep -q 'false' ".ddev/tailscale-router/config/tailscale-private.json"
+  
+  run test -f .ddev/config.tailscale-router.yaml
   assert_success
-
-  # Check public config
-  run grep -q '"AllowFunnel"' ".ddev/tailscale-router/config/tailscale-public.json"
+  
+  run test -f .ddev/web-build/Dockerfile.tailscale-router
   assert_success
-  run grep -q 'true' ".ddev/tailscale-router/config/tailscale-public.json"
+  
+  run test -f .ddev/docker-compose.tailscale-router.yaml
   assert_success
-}
+  
+  # Check that old JSON config files are removed (from the migration)
```

Comment:
You can't really check that it doesn't exist, because test uses fresh add-on all the time.

See my another comment about manual cleanup.